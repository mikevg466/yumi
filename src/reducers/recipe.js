import types from '../actions/types';

export const initialState = {
  name: 'Teriyaki Pulled Pork',
  typeTags: ['Pork', 'Asian'],
  cookMethodTags: ['Slow Cooker'],
  image:
    'https://www.seriouseats.com/recipes/images/2016/06/20160701-sous-vide-pulled-pork-16-1500x1125.jpg',
  ingredients: [
    { type: 'pork', amount: '1 lb' },
    { type: 'teriyaki sauce', amount: '1 cup' },
    { type: 'water', amount: '1/2 cup' },
    { type: 'dry rub', amount: '1 cup' },
    { type: 'water', amount: '1/2 cup' },
  ],
  steps: [
    'pour teriyaki sauce and water into crock pot',
    'place pork in the crock pot',
    'dry rub the top of the pork',
    'cook on medium for 4 hours',
    'shred pork',
    'add pork back in crock pot on warm setting for 1 hour',
  ],
};

/**
 * The main user reducer function to be exported
 * @param  {Object} [state=initialState]
 *    The current user store to be manipulated.
 *    If null, then the default initial state.
 * @param  {Object} action
 *    The reducer action, containing the action type and data to be processed.
 * @return {Object} The updated user state.
 */
export default function recipeReducer(state = initialState, action) {
  switch (action.type) {
    case types.SELECT_RECIPE:
      return action.recipe || initialState;
    default:
      return state;
  }
}
