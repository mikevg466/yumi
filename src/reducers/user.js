import update from 'immutability-helper';
import types from '../actions/types';

export const userInitialState = {
  name: 'world',
  greeting: 'hello',
};

/**
 * The main user reducer function to be exported
 * @param  {Object} [state=userInitialState]
 *    The current user store to be manipulated.
 *    If null, then the default initial state.
 * @param  {Object} action
 *    The reducer action, containing the action type and data to be processed.
 * @return {Object} The updated user state.
 */
export default function userReducer(state = userInitialState, action) {
  switch (action.type) {
    case types.SET_NAME:
      return update(state, {
        name: { $set: action.name },
      });
    case types.SET_GREETING:
      return update(state, {
        greeting: { $set: action.greeting },
      });
    default:
      return Object.assign({}, state);
  }
}
