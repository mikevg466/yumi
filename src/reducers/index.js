import { combineReducers } from 'redux';
import recipe from './recipe';
import user from './user';

// Use ES6 object literal shorthand syntax to define the object shape
const rootReducer = combineReducers({
  recipe,
  user,
});

export default rootReducer;
