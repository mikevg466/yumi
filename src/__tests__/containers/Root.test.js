import React from 'react';
import { shallow } from 'enzyme';
import {
  RootContainer,
  mapStateToProps,
  mapDispatchToProps,
} from '../../containers/Root';
import { userInitialState } from '../../reducers/user';

const mockState = {
  name: userInitialState.name,
  greeting: userInitialState.greeting,
};
const mockDispatchers = {
  setName: jest.fn(),
  setGreeting: jest.fn(),
};
const mockProps = {
  ...mockState,
  ...mockDispatchers,
};

describe('Root Container', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<RootContainer {...mockProps} />);
    expect(wrapper.debug()).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {
    it('outputs the expected state when passed a mock store', () => {
      expect(mapStateToProps({ user: userInitialState })).toEqual(mockState);
    });
  });

  describe('mapDispatchToProps', () => {
    expect(Object.keys(mapDispatchToProps)).toEqual(
      Object.keys(mockDispatchers)
    );
  });
});
