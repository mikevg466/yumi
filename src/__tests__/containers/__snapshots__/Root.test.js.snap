// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root Container renders correctly 1`] = `"<RootComponent name=\\"world\\" greeting=\\"hello\\" setName={[Function: mockConstructor]} setGreeting={[Function: mockConstructor]} />"`;
